File: java/majority_element.java
public class Solution {
    public int majorityElement(int[] nums) {
        int n=0, count=0;
        
        for (int i=0; i<nums.length; i++) {
            if (count==0) {
                n=nums[i];
                count++;
            } else {
                if (nums[i]==n) count++;
                else count--;
            }
        }
        return n;
    }
}

File: python/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        cand, cnt=0, 0
        for n in nums:
            if n==cand:
                cnt+=1
            elif cnt==0:
                cand=n
                cnt+=1
            else:
                cnt-=1
        return cand

